AGENTS.md

```diff
@@ -0,0 +1,433 @@
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/package-*
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fix.patch
+# ... rest of build
+"""
+```
+
+### Platform Filtering
+
+```julia
+# Only 64-bit platforms
+platforms = filter(p -> arch(p) != "i686" && arch(p) != "armv7l", supported_platforms())
+
+# Expand C++ string ABI variants
+platforms = expand_cxxstring_abis(platforms)
+
+# Specific platforms
+platforms = [Platform("x86_64", "linux"), Platform("x86_64", "macos")]
+```
+
+### Setting Environment Variables
+
+```bash
+export CPPFLAGS="-I${includedir}"
+export LDFLAGS="-L${libdir}"
+export PKG_CONFIG_PATH="${prefix}/lib/pkgconfig:${PKG_CONFIG_PATH}"
+```
+
+### Multiple Source Directories
+
+```bash
+cd ${WORKSPACE}/srcdir
+mkdir build && cd build
+cmake ../package-source
+```
+
+## Testing Locally
+
+```bash
+julia --project=/path/to/Yggdrasil build_tarballs.jl --verbose --debug
+```
+
+Use `--debug` to get interactive shell on failure.
+
+**Note**: On macOS, you need Docker installed for local testing.
+
+## Common Issues
+
+### Can't find headers/libraries
+
+Add `-I${includedir}` to `CPPFLAGS` or `CFLAGS`/`CXXFLAGS`
+Add `-L${libdir}` to `LDFLAGS`
+
+### Old `config.sub`/`config.guess`
+
+Call `update_configure_scripts` before `./configure`
+
+### Foreign executable error
+
+Can't run cross-compiled executables. Patch build system or build tools natively.
+
+### Missing shared library on PowerPC
+
+Regenerate configure: `autoreconf -vi` before `./configure`
+
+### Link errors about unrecognized relocations
+
+Increase `preferred_gcc_version` to match dependencies
+
+## File Organization
+
+```text
+PackageName/
+├── build_tarballs.jl
+└── bundled/
+    └── patches/
+        ├── fix1.patch
+        └── fix2.patch
+```
+
+Place in alphabetical directory: `X/XYZ/build_tarballs.jl`
+
+## Commit Messages
+
+Format: `[PackageName] Brief description`
+
+Examples:
+
+- `[Zlib] Update to v1.3.1`
+- `[CMake] Add OpenSSL dependency`
+- `[FFMPEG] Fix build on FreeBSD`
+
+Use `[skip build]` to publish JLL without rebuilding (for metadata-only changes like compat bounds).
+
+## Testing JLL Packages Before Merging
+
+Before merging a Yggdrasil PR, you should test the generated JLL package locally to ensure it works correctly. This avoids automatic registration of broken packages in the General registry.
+
+### Step 0: Speed Up Testing (Optional)
+
+For faster local testing, build only for your current platform by filtering the platforms list. Add this line after the `platforms = [...]` definition:
+
+```julia
+# For local testing, only build for the current platform
+platforms = filter(p -> Sys.isapple() && arch(p) == "aarch64", platforms)  # macOS ARM64
+# platforms = filter(p -> Sys.isapple() && arch(p) == "x86_64", platforms)  # macOS x86_64
+# platforms = filter(p -> Sys.islinux() && arch(p) == "x86_64", platforms)  # Linux x86_64
+# platforms = filter(p -> Sys.iswindows() && arch(p) == "x86_64", platforms)  # Windows x86_64
+```
```

Comment:
IMO, it is better just to pass your local platform to the build call, e.g.,
```
julia --project=/path/to/Yggdrasil build_tarballs.jl --verbose --debug x86_64-linux-gnu
```

instead of doing a lot of these steps.